### ISSUE TITLE
Simple ODE demo problem question about boundary condtions

### ISSUE THREAD

**Issue by jwson302** [lululxvi/deepxde#1170]
This is a question about the demo problem: https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/ode.system.html

The following is a modified code of the demo shown in the website. The only difference is that the domain is [pi/2, pi] rather than [0, 10], and the initial conditions are given as y1(pi/2)=1, y2(pi)=-1, while it was originally y1(0)=0, y2(0)=1.

```
"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch, jax, paddle"""
import deepxde as dde
import numpy as np
from deepxde.backend import tf


def ode_system(x, y):
    """ODE system.
    dy1/dx = y2
    dy2/dx = -y1
    """
    # Most backends
    y1, y2 = y[:, 0:1], y[:, 1:]
    dy1_x = dde.grad.jacobian(y, x, i=0)
    dy2_x = dde.grad.jacobian(y, x, i=1)
    # Backend jax
    # y_val, y_fn = y
    # y1, y2 = y_val[:, 0:1], y_val[:, 1:]
    # dy1_x, _ = dde.grad.jacobian(y, x, i=0)
    # dy2_x, _ = dde.grad.jacobian(y, x, i=1)
    return [dy1_x - y2, dy2_x + y1]



def boundary1(x, on):
    return np.isclose(x[0], np.pi/2)

def boundary2(x, on):
    return np.isclose(x[0], np.pi)


def func(x):
    """
    y1 = sin(x)
    y2 = cos(x)
    """
    return np.hstack((np.sin(x), np.cos(x)))


geom = dde.geometry.TimeDomain(np.pi/2, np.pi)
ic1 = dde.icbc.IC(geom, lambda x: 1, boundary1, component=0)
ic2 = dde.icbc.IC(geom, lambda x: -1, boundary2, component=1)
data = dde.data.PDE(geom, ode_system, [ic1, ic2], 100, 2, solution=func, num_test=100)

layer_size = [1] + [50] * 3 + [2]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001, metrics=["l2 relative error"])
losshistory, train_state = model.train(iterations=20000)

dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

![blob](https://user-images.githubusercontent.com/109210202/222347255-4d772b5f-7981-4140-844c-4245e814f2b1.jpg)

As one can see, the model predicts the sin, cos functions very poorly. 

Is there something wrong with my code? I tried using larger networks, train for more iterations, increase the amount of residual points, changing loss weights, but was unsuccesful at reaching the loss values ~10^-3, where in the demo it was reached quite easily.

Or is it that PINN just doesn't work well for some cases?
